### PR TITLE
DOC-2401: Add integration docs for Azure OpenAI

### PR DESCRIPTION
**Pull request by ltrouton**
Ticket: DOC-2401

Site: [Staging branch](http://docs-feature-71-doc-2401.staging.tiny.cloud/docs/tinymce/latest/ai-azure)

Changes:
*  Add integration docs for Azure OpenAI

Pre-checks:
- [X] Branch prefixed with `feature/<version>/`, `hotfix/<version>/`, `staging/<version>/`, or `release/<version>/`
- [X] `modules/ROOT/nav.adoc` has been updated `(if applicable)`
- [X] Files has been included where required `(if applicable)`
- [X] Files removed have been deleted, not just excluded from the build `(if applicable)`
- [X] Files added for `New product features`, and included a `release note` entry.

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -291,6 +291,7 @@
 *** xref:advanced-typography.adoc[Advanced Typography]
 *** xref:ai.adoc[AI Assistant]
 **** xref:ai-proxy.adoc[AI proxy server reference guide]
+**** xref:ai-azure.adoc[Azure AI integration guide]
```
Comment (FarzadHayat) [inline modules/ROOT/nav.adoc:294]:
```suggestion
**** xref:ai-azure.adoc[Azure AI integration guide]
**** xref:ai-proxy.adoc[AI proxy server reference guide]
```

Comment (FarzadHayat) [reply modules/ROOT/nav.adoc:294]: swap order as we are placing ai-proxy.adoc last in the list

File: modules/ROOT/pages/ai-azure.adoc [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+Before beginning, make sure an OpenAI model has been set up in Azure. The model's endpoint URL and API key are required for integration. For more information, see the https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line%2Cpython-new&pivots=rest-api[Quick Start Guide] provided by Azure.
+
+A proxy is required if making `fetch` calls directly to the Azure OpenAI REST API as otherwise CORS errors arise. In general, using a proxy is highly recommended even if network errors did not occur. See the xref:ai-proxy.adoc[AI Proxy Server reference guide] for information on how to setup a proxy server for use with the {pluginname}.
+
+NOTE: The xref:ai-proxy.adoc[AI Proxy Server reference guide] is, as its name notes, a reference. There is no single proxy server setup that is right or correct for all circumstances and other setups may be better for your use-case.
+
+== String response
+
+This example demonstrates how to integrate the {pluginname} plugin with the https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions[Azure OpenAI Service Chat completions API] to generate a string response.
+
```
Comment (FarzadHayat) [inline modules/ROOT/pages/ai-azure.adoc:33]:
```suggestion

include::partial$misc/admon-ai-proxy.adoc[]

```

File: modules/ROOT/pages/ai-azure.adoc [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  plugins: 'ai',
+  toolbar: 'aidialog aishortcuts',
+  ai_request
+});
+----
+
+== Streaming response
+
+This example demonstrates how to integrate the {pluginname} plugin with the https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions[Azure OpenAI Service Chat completions API] to generate streaming responses.
+
```
Comment (FarzadHayat) [inline modules/ROOT/pages/ai-azure.adoc:82]:
```suggestion

include::partial$misc/admon-ai-proxy.adoc[]

```

**Review by kemister85 — APPROVED** [review]
LGTM

File: modules/ROOT/pages/ai-azure.adoc [hunk 4]
```diff
@@ -0,0 +1,215 @@
+= Azure OpenAI Service integration guide
+:pluginname: AI Assistant
+:plugincode: ai
+:plugincategory: premium
+:navtitle: Azure OpenAI Service integration guide
```
Comment (MitchC1999) [inline modules/ROOT/pages/ai-azure.adoc:5]:
```suggestion
:navtitle: Azure AI integration guide
```
navtitle should match the nav link

File: modules/ROOT/pages/ai-azure.adoc [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+include::partial$misc/admon-ai-pricing.adoc[]
+
+== Introduction
+
+This guide provides instructions for integrating the {pluginname} plugin using the https://learn.microsoft.com/en-us/azure/ai-services/openai/[Azure OpenAI Service] in {productname}.
+
+Azure OpenAI Service provides REST API access to OpenAI's language models with Azure specific features such as private networking, regional availability, and responsible AI content filtering.
+
+Azure OpenAI co-develops the APIs with OpenAI. Therefore, the following examples should be compatible if using Azure OpenAI REST API endpoints or OpenAI API endpoints.
```
Comment (MitchC1999) [inline modules/ROOT/pages/ai-azure.adoc:18]: The OpenAI integration is being separated out to its own page now, and credentials are non-transferrable, so we probably don't need to mention this. Future updates could also make them incompatible.

File: modules/ROOT/pages/ai-azure.adoc (hunk 2 above)
Comment (MitchC1999) [reply modules/ROOT/pages/ai-azure.adoc:33]: This should go above the `== String response` heading

File: modules/ROOT/pages/ai-azure.adoc (hunk 3 above)
Comment (MitchC1999) [reply modules/ROOT/pages/ai-azure.adoc:82]: We only need to include this once per page.

File: modules/ROOT/pages/ai-azure.adoc [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+      // Use microsoft's fetch-event-source library to work around the 2000 character limit
+      // of the browser `EventSource` API, which requires query strings
+      return fetchApi
+      .then(fetchEventSource =>
+        fetchEventSource(AZURE_OPENAI_ENDPOINT, {
+          ...azureOpenAiOptions,
+          openWhenHidden: true,
+		      onopen,
+          onmessage,
```
Comment (MitchC1999) [inline modules/ROOT/pages/ai-azure.adoc:192]:
```suggestion
          openWhenHidden: true,
          onopen,
          onmessage,
```
Just check indentation

File: modules/ROOT/pages/ai-azure.adoc (hunk 2 above)
Comment (FarzadHayat) [reply modules/ROOT/pages/ai-azure.adoc:33]: yes, i will open a new suggestion.

File: modules/ROOT/pages/ai-azure.adoc [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+To learn more about the difference between string and streaming responses, see xref:ai.adoc#the-respondwith-object[The `respondWith` object] on the plugin page.
+
+== Prerequisites
+
+Before beginning, make sure an OpenAI model has been set up in Azure. The model's endpoint URL and API key are required for integration. For more information, see the https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line%2Cpython-new&pivots=rest-api[Quick Start Guide] provided by Azure.
+
+A proxy is required if making `fetch` calls directly to the Azure OpenAI REST API as otherwise CORS errors arise. In general, using a proxy is highly recommended even if network errors did not occur. See the xref:ai-proxy.adoc[AI Proxy Server reference guide] for information on how to setup a proxy server for use with the {pluginname}.
+
+NOTE: The xref:ai-proxy.adoc[AI Proxy Server reference guide] is, as its name notes, a reference. There is no single proxy server setup that is right or correct for all circumstances and other setups may be better for your use-case.
+
```
Comment (FarzadHayat) [inline modules/ROOT/pages/ai-azure.adoc:29]:
```suggestion

include::partial$misc/admon-ai-proxy.adoc[]

```

**Review by FarzadHayat — APPROVED** [review]

File: modules/ROOT/pages/ai-azure.adoc [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+== String response
+
+This example demonstrates how to integrate the {pluginname} plugin with the https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#chat-completions[Azure OpenAI Service Chat completions API] to generate a string response.
+
+[source,js]
+----
+// This example stores the API key in the client side integration. This is not recommended for any purpose.
+// Instead, an alternate method for retrieving the API key should be used.
+const AZURE_OPENAI_API_KEY = '<INSERT_API_KEY_HERE>';
+const AZURE_OPENAI_ENDPOINT = '<INSERT_ENDPOINT_URL_HERE>'; // e.g. https://{your-resource-name}.openai.azure.com/openai/deployments/{deployment-id}/chat/completions?api-version={api-version}
```
Comment (MitchC1999) [inline modules/ROOT/pages/ai-azure.adoc:35]:
```suggestion
const AZURE_OPENAI_ENDPOINT = '<INSERT_ENDPOINT_URL_HERE>'; // e.g. https://<INSERT_RESOURCE_NAME_HERE>.openai.azure.com/openai/deployments/<INSERT_DEPLOYMENT_ID_HERE>/chat/completions?api-version=<INSERT_API_VERSION_HERE>
```
If these are variables, either show that they need to be replaced here, or declare them as separate variables before this line.

File: modules/ROOT/pages/ai-azure.adoc [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+const ai_request = (request, respondWith) => {
+  const azureOpenAiOptions = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'api-key': AZURE_OPENAI_API_KEY
```
Comment (MitchC1999) [inline modules/ROOT/pages/ai-azure.adoc:42]:
```suggestion
      'api-key': `Bearer ${AZURE_OPENAI_API_KEY}`
```
Just want confirmation that this is not the correct format.

Comment (ltrouton) [reply modules/ROOT/pages/ai-azure.adoc:42]: The Azure quick start guide indicates using an `api-key` header so have just followed that: https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=command-line%2Cpython-new&pivots=rest-api#rest-api

File: modules/ROOT/pages/ai-azure.adoc [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+
+include::partial$misc/admon-ai-pricing.adoc[]
+
+== Introduction
+
+This guide provides instructions for integrating the {pluginname} plugin using the https://learn.microsoft.com/en-us/azure/ai-services/openai/[Azure OpenAI Service] in {productname}.
+
+Azure OpenAI Service provides REST API access to OpenAI's language models with Azure specific features such as private networking, regional availability, and responsible AI content filtering.
+
+To learn more about the difference between string and streaming responses, see xref:ai.adoc#the-respondwith-object[The `respondWith` object] on the plugin page.
```
Comment (ShiridiGandham) [inline modules/ROOT/pages/ai-azure.adoc:18]:
```suggestion
To learn more about the difference between string and streaming responses, see the xref:ai.adoc#the-respondwith-object[`respondWith` object] on the plugin page.
```

File: modules/ROOT/pages/ai-azure.adoc [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+[source,js]
+----
+const fetchApi = import("https://unpkg.com/@microsoft/fetch-event-source@2.0.1/lib/esm/index.js").then(module => module.fetchEventSource);
+
+// This example stores the API key in the client side integration. This is not recommended for any purpose.
+// Instead, an alternate method for retrieving the API key should be used.
+const AZURE_OPENAI_API_KEY = '<INSERT_API_KEY_HERE>';
+const AZURE_OPENAI_ENDPOINT = '<INSERT_ENDPOINT_URL_HERE>'; // e.g. https://{your-resource-name}.openai.azure.com/openai/deployments/{deployment-id}/chat/completions?api-version={api-version}
+
+tinymce.init({
```
Comment (shanmen-tiny) [inline modules/ROOT/pages/ai-azure.adoc:88]: We have two `tinymce.init` here. Should we just use `ai_request` here instead?

**Review by shanmen-tiny — APPROVED** [review]

File: modules/ROOT/pages/ai-azure.adoc (hunk 11 above)
Comment (ltrouton) [reply modules/ROOT/pages/ai-azure.adoc:88]: Oh, I am not sure what happened here. I thought I just copied the code example from one of the original code snippets in the AI plugin docs page and adjusted some of the variables.

I will fix this up 👍

**Review by FarzadHayat — APPROVED** [review]
LGTM ✅

File: modules/ROOT/nav.adoc [hunk 12]
```diff
@@ -293,6 +293,7 @@
 **** xref:ai-openai.adoc[OpenAI ChatGPT integration guide]
 **** xref:ai-bedrock.adoc[Amazon Bedrock integration guide]
 **** xref:ai-gemini.adoc[Google Gemini integration guide]
+**** xref:ai-azure.adoc[Azure AI integration guide]
```
Comment (FarzadHayat) [inline modules/ROOT/nav.adoc:296]:
```suggestion
**** xref:ai-azure.adoc[Azure AI integration guide]
**** xref:ai-bedrock.adoc[Amazon Bedrock integration guide]
**** xref:ai-gemini.adoc[Google Gemini integration guide]
```

File: modules/ROOT/pages/ai-azure.adoc [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+
+This guide provides instructions for integrating the {pluginname} plugin using the https://learn.microsoft.com/en-us/azure/ai-services/openai/[Azure OpenAI Service] in {productname}.
+
+Azure OpenAI Service provides REST API access to OpenAI's language models with Azure specific features such as private networking, regional availability, and responsible AI content filtering.
+
+To learn more about the difference between string and streaming responses, see the xref:ai.adoc#the-respondwith-object[`respondWith` object] on the plugin page.
+
+== Prerequisites
+
+Before beginning, make sure an OpenAI model has been set up in Azure. The model's endpoint URL and API key are required for integration. For more information, see the https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line%2Cpython-new&pivots=rest-api[Quick Start Guide] provided by Azure.
```
Comment (kemister85) [inline modules/ROOT/pages/ai-azure.adoc:22]: Should we consider opening external links in a new tab?
```suggestion
Before beginning, make sure an OpenAI model has been set up in Azure. The model's endpoint URL and API key are required for integration. For more information, see the https://learn.microsoft.com/en-us/azure/ai-services/openai/chatgpt-quickstart?tabs=command-line%2Cpython-new&pivots=rest-api[Quick Start Guide^] provided by Azure.
```

**Review by MitchC1999 — APPROVED** [review]

**Pull request merged**